File: backend/TaskFlow.Model/Entities/TaskItem.cs
namespace TaskFlow.Model.Entities;

public class TaskItem
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset? DueAt { get; set; }
    public string Priority { get; set; } = "medium";
    public string Category { get; set; } = "GENERAL";
    public string? Description { get; set; }
    public bool Done {get; set; }
}

File: frontend/src/app/models/task-item.ts
export type TaskPriorityLevel = 'high' | 'medium' | 'low';

export interface TaskItem {
  id: number;
  title: string;
  createdAt: string;
  dueAt: string | null;
  priority: TaskPriorityLevel;
  category: string | null;
  description: string | null;
  done: boolean;
}
